dask/optimization.py

```diff
@@ -937,10 +938,12 @@ class SubgraphCallable:
 
     __slots__ = ("dsk", "outkey", "inkeys", "name")
 
-    def __init__(self, dsk, outkey, inkeys, name="subgraph_callable"):
+    def __init__(self, dsk, outkey, inkeys, name=None):
         self.dsk = dsk
         self.outkey = outkey
         self.inkeys = inkeys
+        if name is None:
+            name = tokenize(dsk, outkey, inkeys)
```

Comment:
tokenizing the whole dsk dict is potentially exceptionally expensive and unnecessary